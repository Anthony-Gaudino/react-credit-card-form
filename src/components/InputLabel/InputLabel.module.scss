.inputLabel {
  display: block;
  width: 100%;
  font-size: 0.8rem;

  &--hidden {
    font-size: 0;
  }
}
