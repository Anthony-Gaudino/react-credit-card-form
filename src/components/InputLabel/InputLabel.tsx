import clsx from "clsx";

import styles from "./InputLabel.module.scss";

interface InputLabelProps {
  /**
   * Label text.
   */
  label: string;
  /**
   * If `true` the label is hidden.
   */
  hidden?: boolean;
  /**
   * Label children.
   */
  children?: React.ReactNode;
}

/**
 * A HTML label that can have an input element as children.
 */
export const InputLabel = (props: InputLabelProps): JSX.Element => {
  const { label, children, hidden = false } = props;

  const classes = clsx({
    [styles.inputLabel]: true,
    [styles["inputLabel--hidden"]]: hidden,
  });

  return (
    <label className={classes}>
      {label}
      {children}
    </label>
  );
};
