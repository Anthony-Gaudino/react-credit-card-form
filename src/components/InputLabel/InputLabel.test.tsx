import { render, screen } from "@testing-library/react";

import InputLabel from ".";

describe("Test InputLabel component", () => {
  test("the label is correctly rendered", () => {
    const { asFragment } = render(<InputLabel label="A label" />);

    const label = screen.getByText("A label");

    expect(label).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
