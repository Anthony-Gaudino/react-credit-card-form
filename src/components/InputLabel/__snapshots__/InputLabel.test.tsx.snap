// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputLabel component the label is correctly rendered 1`] = `
<DocumentFragment>
  <label
    class="inputLabel"
  >
    A label
  </label>
</DocumentFragment>
`;
